// @dicebear/rings/lib/utils/getComponents.js
/**
 * Do not change this file manually! This file was generated with the "Dicebear Exporter"-Plugin for Figma.
 *
 * Plugin: https://www.figma.com/community/plugin/1005765655729342787
 * File: https://www.figma.com/file/sK2mrFhCZwQu1qw4WqOBEF
 */
import { pickComponent } from './pickComponent.js';
export function getComponents({ prng, options }) {
    var _a, _b, _c, _d, _e, _f;
    const ringComponent = pickComponent({
        prng,
        group: 'ring',
        values: options.ring,
        width: 100,
        height: 100,
        rotation: ((_a = options.ringRotation) === null || _a === void 0 ? void 0 : _a.length) ? options.ringRotation : [0],
        offsetX: [0],
        offsetY: [0],
    });
    const ringOneComponent = pickComponent({
        prng,
        group: 'ringOne',
        values: options.ringOne,
        width: 100,
        height: 100,
        rotation: ((_b = options.ringOneRotation) === null || _b === void 0 ? void 0 : _b.length) ? options.ringOneRotation : [0],
        offsetX: [0],
        offsetY: [0],
    });
    const ringTwoComponent = pickComponent({
        prng,
        group: 'ringTwo',
        values: options.ringTwo,
        width: 100,
        height: 100,
        rotation: ((_c = options.ringTwoRotation) === null || _c === void 0 ? void 0 : _c.length) ? options.ringTwoRotation : [0],
        offsetX: [0],
        offsetY: [0],
    });
    const ringThreeComponent = pickComponent({
        prng,
        group: 'ringThree',
        values: options.ringThree,
        width: 100,
        height: 100,
        rotation: ((_d = options.ringThreeRotation) === null || _d === void 0 ? void 0 : _d.length) ? options.ringThreeRotation : [0],
        offsetX: [0],
        offsetY: [0],
    });
    const ringFourComponent = pickComponent({
        prng,
        group: 'ringFour',
        values: options.ringFour,
        width: 100,
        height: 100,
        rotation: ((_e = options.ringFourRotation) === null || _e === void 0 ? void 0 : _e.length) ? options.ringFourRotation : [0],
        offsetX: [0],
        offsetY: [0],
    });
    const ringFiveComponent = pickComponent({
        prng,
        group: 'ringFive',
        values: options.ringFive,
        width: 100,
        height: 100,
        rotation: ((_f = options.ringFiveRotation) === null || _f === void 0 ? void 0 : _f.length) ? options.ringFiveRotation : [0],
        offsetX: [0],
        offsetY: [0],
    });
    return {
        'ring': ringComponent,
        'ringOne': ringOneComponent,
        'ringTwo': ringTwoComponent,
        'ringThree': ringThreeComponent,
        'ringFour': ringFourComponent,
        'ringFive': ringFiveComponent,
    };
}
;
